### ISSUE TITLE
Load params and performing a linear simulation

### ISSUE THREAD

**Issue by akhoubani** [snek5000/snek5000#124]
We would like to perform a linear simulation using an existing base flow file. For doing so, we should be able to load `params`, modify the params and copy a file from an existing non-linear simulation. 

```python
from shutil import copyfile

from snek5000_cbox.solver import Simul
from snek5000.params import load_params

params = load_params(sim_path)
params.nek.general.start_from = "base_flow.restart"

params.nek.problemtype.equation = "incompLinNS"
params.NEW_DIR_RESULTS = True
restart_file = params.output.path_session / "cbox0.f00100"

sim = Simul(params)

copyfile(restart_file, sim.path_run / "base_flow.restart")

sim.make.exec('run_fg', resources={"nproc": 4})
```
This script gives this error:
```python
FileNotFoundError: [Errno 2] No such file or directory: 'examples_cbox/simple/cbox_Ra1.835e+08_80x80_8x8_V1.x1._2022-01-04_10-54-25/params_simul.xml'
```

**Comment by paugier** [comment]
Does this file actually exist? "examples_cbox/simple/cbox_Ra1.835e+08_80x80_8x8_V1.x1._2022-01-04_10-54-25/params_simul.xml"

**Comment by akhoubani** [comment]
using `params = load_params("/home/arman/Sim_data/examples_cbox/simple/cbox_Ra1.835e+08_80x80_8x8_V1.x1._2022-01-04_10-54-25")` I could run the simulation. 

[build.log](https://github.com/exabl/snek5000/files/7816132/build.log)
[cbox.log](https://github.com/exabl/snek5000/files/7816135/cbox.log)

**Comment by akhoubani** [comment]
If I use `nekbmpi cbox 4` in the directory:

```fortran
Initialization successfully completed   0.21642E-01 sec

Starting time loop ...

     DT/DTCFL/DTFS/DTINIT   0.480E-01   0.480E-01   0.000E+00   0.000E+00
Step      1, t= 4.8005548E-02, DT= 4.8005548E-02, C=  2.000 0.0000E+00 0.0000E+00
 no convchp
 no convchp
  
an error occured: dying ...
  
 no convchp
 no convchp
```

**Comment by paugier** [comment]
So the problem for you is the difference between `load_simul` and `load_params`, i.e. the fact that `load_simul` supports using short path without "/home/arman/Sim_data" but no `load_params`? Am I right?

Do you have another problem?

**Comment by akhoubani** [comment]
> So the problem for you is the difference between `load_simul` and `load_params`, i.e. the fact that `load_simul` supports using short path without "/home/arman/Sim_data" but no `load_params`? Am I right?

Yes.

> 
> Do you have another problem?

Yes, there is another problem. It fails.

**Comment by paugier** [comment]
About the first problem, could you please try the branch `snek5000.load_params`?

About the second problem, can you please try to analyze a bit why it fails? And maybe copy/paste here a meaningful part of the error log.

**Comment by akhoubani** [comment]
Yes, I'll let you know.

**Comment by akhoubani** [comment]
Yes. Thanks.

**Comment by akhoubani** [comment]
About the second problem, it gives this error:

```fortran
call usrdat3
 done :: usrdat3

 set initial conditions
 Checking restart options: base_flow.restart                                                                                                                   
 nekuic (1) for ifld            1
 nekuic (1) for ifld            2
 nekuicP           2           1 T
 Reading checkpoint data 
./session_00/base_flow.restart
byte_read() :: fopen failure2!
ERROR: Error reading restart header in mfi_prepare  ierr=  1 
  
an error occured: dying ...
  
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpiexec detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[42969,1],1]
  Exit code:    1
```
We don't have `base_flow.restart` in the `session_00` directory.

**Comment by akhoubani** [comment]
If I use:

```python 
copyfile(restart_file, sim.params.output.path_session / "base_flow.restart")
```
it gives this error:
```fortran
Initialization successfully completed   0.22427E-01 sec

Starting time loop ...

     DT/DTCFL/DTFS/DTINIT   0.480E-01   0.480E-01   0.000E+00   0.000E+00
Step      1, t= 4.8005548E-02, DT= 4.8005548E-02, C=  2.000 0.0000E+00 0.0000E+00
 no convchp
 no convchp
 no convchp
 no convchp
  
an error occured: dying ...
```

**Comment by akhoubani** [comment]
In the Nek source code [here](https://github.com/Nek5000/Nek5000/blob/f6844aba474905da20a188398b5e3d7e7c4b4ef9/core/perturb.f#L679) we have the `no convchp`.

**Comment by akhoubani** [comment]
Yes, by using `params.nek.general.extrapolation = "standard"` it runs. The other issue could be how to change history points subroutine to have perturbation field in `cbox.his`, not the base flow.

**Comment by paugier** [comment]
Good that you finally managed to launch the linear simulation.

> The other issue could be how to change history points subroutine to have perturbation field in cbox.his, not the base flow.

This is a completely different problem. Please fill another issue.

And please can you summarize if there are still things to be fixed for starting a linear simulation from an old simulation?

Maybe we just need to add an example?

**Comment by akhoubani** [comment]
I'll add an example to the cbox.

I don't know if you would like to include a restart option for linear simulation in `util/restart`. Now with a script that loads `params` it works.

**Comment by paugier** [comment]
@akhoubani could you please check if this issue can be closed? Is there still something to do?

**Comment by akhoubani** [comment]
@paugier yes, it can be closed.
